Gems/ROS2/Code/Source/EffortSensor/ROS2EffortSensorComponent.cpp

```diff
@@ -0,0 +1,108 @@
+        for (auto type : CompatibleJointComponents ){
+            auto foundComponents = m_entity->FindComponents(type);
+            AZStd::move(foundComponents.begin(), foundComponents.end(), AZStd::back_inserter(components));
+        }
+        return components;
+    }
+
+    void ROS2EffortSensorComponent::Activate()
+    {
+        AZ::Entity::ComponentArrayType componentPtrVec = findCompatibleJointComponents();
+        AZ_Warning("ROS2EffortSensorComponent", componentPtrVec.size()>0, "ROS2EffortSensorComponent can work with only one joint component.");
+        AZ_Warning("ROS2EffortSensorComponent", componentPtrVec.size()==0, "ROS2EffortSensorComponent needs compatible joint");
+        if (componentPtrVec.size()>0){
+            m_jointId = AZ::EntityComponentIdPair(m_entity->GetId(), componentPtrVec.front()->GetId());
+        }
+        ROS2SensorComponent::Activate();
+
+        auto ros2Node = ROS2Interface::Get()->GetNode();
+        AZ_Assert(m_sensorConfiguration.m_publishersConfigurations.size() == 1, "Invalid configuration of publishers for Effort sensor");
+
+        const auto publisherConfig = m_sensorConfiguration.m_publishersConfigurations[Internal::kWrenchMsgType];
+        const auto fullTopic = ROS2Names::GetNamespacedName(GetNamespace(), publisherConfig.m_topic);
+        m_effortPublisher = ros2Node->create_publisher<geometry_msgs::msg::WrenchStamped>(fullTopic.data(), publisherConfig.GetQoS());
+    }
+    ROS2EffortSensorComponent::ROS2EffortSensorComponent()
+    {
+        TopicConfiguration tc;
+        const AZStd::string type = Internal::kWrenchMsgType;
+        tc.m_type = type;
+        tc.m_topic = "effort";
+        m_sensorConfiguration.m_frequency = 10;
+        m_sensorConfiguration.m_publishersConfigurations.insert(AZStd::make_pair(type, tc));
+    }
+
+    void ROS2EffortSensorComponent::FrequencyTick()
+    {
+        AZStd::pair<AZ::Vector3, AZ::Vector3> wrench{0,0};
+        PhysX::JointRequestBus::EventResult(wrench, m_jointId, &PhysX::JointRequests::getWrench);
+
+        auto* ros2Frame = Utils::GetGameOrEditorComponent<ROS2FrameComponent>(GetEntity());
+
+
+        geometry_msgs::msg::WrenchStamped msg;
+        msg.header.stamp = ROS2Interface::Get()->GetROSTimestamp();
+        msg.header.frame_id = ros2Frame->GetParentFrameID().c_str();
+
+        msg.wrench.torque.x = wrench.second.GetX();
+        msg.wrench.torque.y = wrench.second.GetY();
+        msg.wrench.torque.z = wrench.second.GetZ();
+
+        msg.wrench.force.x = wrench.first.GetX();
+        msg.wrench.force.y = wrench.first.GetY();
+        msg.wrench.force.z = wrench.first.GetZ();
+
+        m_effortPublisher->publish(msg);
+    }
+
+    void ROS2EffortSensorComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("PhysicsRigidBodyService"));
+//        required.push_back(AZ_CRC_CE("PhysicsJointService"));
```

Comment:
nit: Can remove this commented out code if/when it's not needed

Reply:
Problem is that the runtime component does not provide this service.

Reply:
I see, so this is here for information purposes? If that's the case it would be good to leave a comment why this commented line is here for reference. Maybe it's something we could then add in future.

Reply:
I've removed it from the committed code, we should consider adjusting provided services in physx gem, I will open issue on that.

Reply:
That sounds like a good plan, thanks @michalpelka